go/cron/flush_radius_audit_log_job.go

```diff
@@ -0,0 +1,263 @@
+package maint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/inverse-inc/go-utils/log"
+	"github.com/inverse-inc/packetfence/go/pfconfigdriver"
+	"github.com/inverse-inc/packetfence/go/redisclient"
+)
+
+type FlushRadiusAuditLogJob struct {
+	Task
+	Batch   int
+	Timeout time.Duration
+	redis   *redis.Client
+}
+
+func NewFlushRadiusAuditLogJob(config map[string]interface{}) JobSetupConfig {
+	return &FlushRadiusAuditLogJob{
+		Task:    SetupTask(config),
+		Batch:   int(config["batch"].(float64)),
+		Timeout: time.Duration((config["timeout"].(float64))) * time.Second,
+		redis:   redisClient(),
+	}
+}
+
+func redisClient() *redis.Client {
+	ctx := context.Background()
+	pfconfigdriver.PfconfigPool.AddStruct(ctx, &redisclient.Config)
+	var network string
+	if redisclient.Config.RedisArgs.Server[0] == '/' {
+		network = "unix"
+	} else {
+		network = "tcp"
+	}
+
+	return redis.NewClient(&redis.Options{
+		Addr:    redisclient.Config.RedisArgs.Server,
+		Network: network,
+	})
+}
+
+func (j *FlushRadiusAuditLogJob) Run() {
+	start := time.Now()
+	rows_affected := 0
+	i := 0
+	ctx := context.Background()
+	for {
+		i++
+		data := j.redis.LPopCount(ctx, "", j.Batch)
+		if err := data.Err(); err != nil {
+			//logError(err.Error())
+			return
+		}
+
+		a := data.Val()
+		if len(a) == 0 {
+			break
+		}
+
+		rows_affected += len(a)
+		jsonStr := "[" + strings.Join(a, ",") + "]"
+		var entries [][]interface{} = make([][]interface{}, len(a))
+		json.Unmarshal([]byte(jsonStr), &entries)
+		j.flushLogs(entries)
+		if time.Now().Sub(start) > j.Timeout {
+			break
+		}
+	}
+	if rows_affected > -1 {
```

Comment:
do we need to check this ?

Reply:
You are correct that check is not useful.